src/SmoFacade/Listener.cs

```diff
@@ -134,15 +135,27 @@ private static string AgListenerName(string dataSource)
       string credentialName)
     {
       var parts = agInstanceName.Split('\\');
-      if(parts.Length == 1)
-        connBuilder.DataSource = Dns.GetHostEntry(agInstanceName).HostName;
-      else if(parts.Length == 2)
-        // NamedInstances: chop instance name, resolve DNS, slap instance name back on!
-        connBuilder.DataSource = $"{Dns.GetHostEntry(parts[0]).HostName}\\{parts[1]}";
-      else
-        throw new ArgumentException($"agInstanceName param {agInstanceName} cannot be resolved by DNS");
+            if (parts.Length == 1)
+                connBuilder.DataSource = ResolveDnsAndGetHostName(connBuilder.DataSource, agInstanceName);
+            else if (parts.Length == 2)
+                // NamedInstances: chop instance name, resolve DNS, slap instance name back on!
+                connBuilder.DataSource = $"{ResolveDnsAndGetHostName(connBuilder.DataSource, parts[0])}\\{parts[1]}";
+            else
+                throw new ArgumentException($"agInstanceName param {agInstanceName} cannot be resolved by DNS");
 
       return new Server(connBuilder.ToString(), credentialName);
     }
+
+    private static string ResolveDnsAndGetHostName(string hostUrl, string instance)

```

Comment:
This looks like what @am99 was suggesting, but it's unclear here the difference between the instance and hostUrl. Let's rename and/or document the difference here.

Reply:
Sure have renamed and added some summary comments - let me know if you guys want the wording changed